src/main/java/com/kosherjava/zmanim/hebrewcalendar/TefilaRules.java

```diff
@@ -207,22 +204,18 @@ public boolean isTachanunRecitedShacharis(JewishCalendar jewishCalendar) {
 	 * @see #isTachanunRecitedShacharis(JewishCalendar)
 	 */
 	public boolean isTachanunRecitedMincha(JewishCalendar jewishCalendar) {
-		JewishCalendar tomorrow = new JewishCalendar();
-		tomorrow = (JewishCalendar) jewishCalendar.clone();
+		JewishCalendar tomorrow = (JewishCalendar) jewishCalendar.clone();
 		tomorrow.forward(Calendar.DATE, 1);
-		
-		if (!tachanunRecitedMinchaAllYear
-					|| jewishCalendar.getDayOfWeek() == Calendar.FRIDAY
-					|| ! isTachanunRecitedShacharis(jewishCalendar) 
-					|| (! isTachanunRecitedShacharis(tomorrow) && 
-							!(tomorrow.getYomTovIndex() == JewishCalendar.EREV_ROSH_HASHANA) &&
-							!(tomorrow.getYomTovIndex() == JewishCalendar.EREV_YOM_KIPPUR) &&
-							!(tomorrow.getYomTovIndex() == JewishCalendar.PESACH_SHENI))
-					|| ! tachanunRecitedMinchaErevLagBaomer && tomorrow.getYomTovIndex() == JewishCalendar.LAG_BAOMER) {
-			return false;
-		}
-		return true;
-	}
+
+        return tachanunRecitedMinchaAllYear
+                && jewishCalendar.getDayOfWeek() != Calendar.FRIDAY
+                && isTachanunRecitedShacharis(jewishCalendar)
+                && (isTachanunRecitedShacharis(tomorrow) ||
+                tomorrow.getYomTovIndex() == JewishCalendar.EREV_ROSH_HASHANA ||
+                tomorrow.getYomTovIndex() == JewishCalendar.EREV_YOM_KIPPUR ||
+                tomorrow.getYomTovIndex() == JewishCalendar.PESACH_SHENI)
+                && (tachanunRecitedMinchaErevLagBaomer || tomorrow.getYomTovIndex() != JewishCalendar.LAG_BAOMER);
+    }

```

Comment:
The logic shouldn't have changed but it seems like intellij also took liberties here